### PR TITLE
Many changes, primarily to DOS related functions

### PR DESCRIPTION
**Pull request by AerysL**
This pull requests includes many bug fixes (primarily memory related), formatting changes, documentation improvements, and many other improvements, some of which follow:

- Replace a number of references to evsdata.ifGenEv with a const set to the value
 - Additional consts where appropriate
 - Improve LanDosG to work for standard eigenvalue problems
 - Have LanDos give an error if attempted on a general eigenvalue problem
 - Fix LanDos,LanDosG, and tests in MM_Gen_Landos to support reading more than one pair of matrices from matfile
 - Improve readme to have a better appearance in doxygen

I am submitting this as a pull request for easier review.

### CODE REVIEW

File: EVSL_1.0/TESTS/Gen_MM_Landos/io.c [hunk 1]
```diff
@@ -115,8 +115,8 @@ int read_coo_MM(const char *matfile, int idxin, int idxout, cooMat *Acoo) {
   else
     nnz2 = nnz;
   /*-------- Allocate mem for COO */
-  int* IR = (int *)malloc(nnz2*sizeof(int));
-  int* JC = (int *)malloc(nnz2*sizeof(int));
+  int* IR = (int *)calloc(nnz2,sizeof(int));
+  int* JC = (int *)calloc(nnz2,sizeof(int));
```
Comment (liruipeng) [inline EVSL_1.0/TESTS/Gen_MM_Landos/io.c]: why we need calloc here?

File: EVSL_1.0/TESTS/Landos/io.c [hunk 2]
```diff
@@ -115,8 +115,8 @@ int read_coo_MM(const char *matfile, int idxin, int idxout, cooMat *Acoo) {
   else
     nnz2 = nnz;
   /*-------- Allocate mem for COO */
-  int* IR = (int *)malloc(nnz2*sizeof(int));
-  int* JC = (int *)malloc(nnz2*sizeof(int));
+  int* IR = (int *)calloc(nnz2,sizeof(int));
+  int* JC = (int *)calloc(nnz2,sizeof(int));
   double* VAL = (double *)malloc(nnz2*sizeof(double));
```
Comment (liruipeng) [inline EVSL_1.0/TESTS/Landos/io.c]: why calloc?

**Review by liruipeng — COMMENTED** [review]
looks good! I just have two comments about calloc

File: EVSL_1.0/TESTS/Gen_MM_Landos/io.c (hunk 1 above)
Comment (AerysL) [reply EVSL_1.0/TESTS/Gen_MM_Landos/io.c]: callocs were due to the fact that when I ran a static analysis tool, it detected a route that would result in a reaching lines 161 without having initialized IR,JC with values, but I think that only happens if nnz is 0, at which there would be larger problems.

File: EVSL_1.0/TESTS/Landos/io.c (hunk 2 above)
Comment (AerysL) [reply EVSL_1.0/TESTS/Landos/io.c]: callocs were due to the fact that when I ran a static analysis tool, it detected a route that would result in a reaching lines 161 without having initialized IR,JC with values, but I think that only happens if nnz is 0, at which there would be larger problems.

File: EVSL_1.0/TESTS/MM/io.c [hunk 3]
```diff
@@ -108,8 +108,8 @@ int read_coo_MM(const char *matfile, int idxin, int idxout, cooMat *Acoo) {
   else
     nnz2 = nnz;
   /*-------- Allocate mem for COO */
-  int* IR = (int *)malloc(nnz2*sizeof(int));
-  int* JC = (int *)malloc(nnz2*sizeof(int));
+  int* IR = (int *)calloc(nnz2,sizeof(int));
+  int* JC = (int *)calloc(nnz2,sizeof(int));
```
Comment (liruipeng) [inline EVSL_1.0/TESTS/MM/io.c]: calloc here as well

**Review by liruipeng — COMMENTED** [review]
calloc

**Review by liruipeng — APPROVED** [review]

File: EVSL_1.0/TESTS/MM/io.c (hunk 3 above)
Comment (AerysL) [reply EVSL_1.0/TESTS/MM/io.c]: Fixed, along with a few others.

**Comment by liruipeng** [comment]
merged

**Comment by eigs** [comment]
Quite a few changes.. All seem good. Thanks both for the excellent work.
I will take a look at the DOS for the generalized case..

**Pull request merged**